src/auth/authApp.js

```diff
@@ -84,23 +82,123 @@ export function redirectToSignInWithAuthRequest(authRequest: string = makeAuthRe
     return
   }
 
-  function successCallback() {
-    Logger.info('protocol handler detected')
-    // protocolCheck should open the link for us
+  // Create a unique ID used for this protocol detection attempt.
+  const echoReplyID = Math.random().toString(36).substr(2, 9)
+  const echoReplyKey = `echo-reply-${echoReplyID}`
+
+  // Use localStorage as a reliable cross-window communication method.
+  // Create the storage entry to signal a protocol detection attempt for the
+  // next browser window to check.
+  window.localStorage.setItem(echoReplyKey, 'pending')
+  const cleanUpLocalStorage = () => {
```

Comment:
To avoid cluttering the localStorage right? We can, but then run the risk of breaking an in-progress protocol-ping detection, for the same web app in another tab. Unlikely but possible.
Which error condition in either of these is worse?

We can also add a timestamp to the key's value (currently value is just set to "pending"), and clear out all keys older than 1 minute or so?

Reply:
To avoid re-using echo reply keys by mistake.  Clearing them out if they're e.g. over an hour old should be fine.

Reply:
👍will get this implemented.

Reply:
Yup, looks good :+1: